sympy/matrices/expressions/transpose.py

```diff
@@ -37,6 +37,8 @@ def doit(self, **hints):
         if hints.get('deep', True) and isinstance(arg, Basic):
             arg = arg.doit(**hints)
         try:
+            if isinstance(arg, MatrixExpr) and arg.shape == (1, 1):
+                return arg
```

Comment:
I think that it would suffice to check if the matrix is known to be symmetric by something like `ask(Q.symmetric(arg))`. That is what `refine_transpose()` does.